--- utils.py	2022-03-25 23:32:00.000000000 +0800
+++ utils_mod.py	2022-03-25 23:32:28.000000000 +0800
@@ -96,7 +96,7 @@
     if extra_request_headers is not None:
         params["headers"].update(extra_request_headers)
     # so the connection does not hang indefinitely (read/connection timeout)
-    timeout = params.get("timeout", 60)
+    timeout = 600

     if url.startswith("ftp://"):
         response = urlopen(url, timeout=timeout)
